### PR TITLE
docs(nextly): separate the two access layers a missing user acts on

### PR DESCRIPTION
**Pull request by mobeenabdullah**
`packages/nextly/AGENTS.md` said that enforcing collection and document access needs `overrideAccess: false` plus a `user`, because *"a stored owner rule has nobody to compare against otherwise, so it is skipped rather than failed."*

Executed against the real service, an `owner-only` rule with no user returns:

```
{ "allowed": false, "reason": "Authentication required" }
```

It is **denied**, not skipped.

## Both behaviours are real, at different layers

The sentence attributed the skip to the layer that denies.

- The coarse **RBAC gate** in `collection-access-service.ts` needs a user in order to have permissions to check. Every branch is guarded on `user`, so without one it does not run. *This* is what a missing user skips.
- The **stored rules** run either way, and that is where an anonymous caller is refused.

So an anonymous caller is stopped by the rule rather than waved past the gate, which is the opposite of what a reader would conclude. This is the first file an agent session reads, so a wrong claim here is read as true and acted on.

## The rule-less default, added

It is the other half of the same question and it surprises: a collection with **no rule** for an operation is public, since `evaluateAccess` returns `allowed: true` when `rules?.[operation]` is absent. Two cases fail closed on top of that, both in `collection-access-service.ts`: `publish`/`unpublish` with no user and no explicit rule, and `routeAuthorized` with no user.

## Verification

Both clauses are pinned by tests rather than left as prose, because I established them by running the service rather than reading it. `AccessControlService without a user` covers the denial and the rule-less default; 15 tests pass in that file.

One limit worth stating plainly: these pin the **code**. If the behaviour changes they fail. Nothing stops the doc from drifting again on its own, and matching for phrases like "skipped rather than failed" would be the brittle kind of guard that has produced several false positives on #1566. Making AGENTS.md claims machine-checkable in general needs each load-bearing claim to carry a named test and a guard that fails when one has none, which is a design change rather than part of this fix.

### CODE REVIEW

**Comment by mobeenabdullah** [comment]
@codex review

File: packages/nextly/src/services/access/access-control-service.test.ts [hunk 1]
```diff
@@ -56,6 +56,39 @@ describe("AccessControlService role-based access (OR-membership)", () => {
   });
 });
 
+describe("AccessControlService without a user", () => {
+  // `packages/nextly/AGENTS.md` states these two outcomes as the thing agents get
+  // wrong about enforcement, so they are pinned here: the doc said an owner rule
+  // with no user was skipped, and the service denies it. A claim in that file is
+  // read before any work happens, so it is worth a test rather than a reading.
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/services/access/access-control-service.test.ts]: Fixed in c29f6f0. You're right, and it is the same violation this PR is correcting elsewhere: the comment recorded what the documentation had said and why it changed rather than what the assertion distinguishes.

It describes the behaviour now, and points at the file that exercises the layer pairing instead of narrating why the pairing matters.

File: packages/nextly/src/services/access/access-control-service.test.ts [hunk 2]
```diff
@@ -56,6 +56,39 @@ describe("AccessControlService role-based access (OR-membership)", () => {
...
+  const service = new AccessControlService();
+
+  it("DENIES an owner-only rule rather than skipping it", async () => {
+    // Nobody to compare an owner against is a refusal, not an absence of
+    // opinion. What a missing user skips is the coarse RBAC gate, one layer up,
+    // which needs a user in order to have permissions to check.
+    const result = await service.evaluateAccess(
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/nextly/src/services/access/access-control-service.test.ts:67]: Fixed in c29f6f0, and this was the better finding of the two. The claim is about how two layers combine and the test proved the leaf, so it would have stayed green through exactly the regression it was written to prevent.

`collection-access-anonymous.test.ts` drives `checkCollectionAccess` with the real `AccessControlService` and an RBAC spy, so both halves are observable in one assertion: `rbac.checkAccess` is never called, and the request is refused anyway.

Verified by causing the regression you describe. Adding `|| !user` to the `deferStoredRuleEval` early return fails two cases in the new file and **none** in the leaf test, which is your point demonstrated rather than accepted.

Three cases, since the neighbours answer differently and a test that denied everything would look correct on the first alone: the owner-only denial, the rule-less default that allows, and publish with no user and no rule, which this service fails closed on top of the leaf.

**Comment by mobeenabdullah** [comment]
@codex review
